crates/editor/src/items.rs

```diff
@@ -702,12 +703,25 @@ impl Item for Editor {
         }
     }
 
-    fn save(&mut self, project: Model<Project>, cx: &mut ViewContext<Self>) -> Task<Result<()>> {
+    fn save(
+        &mut self,
+        project: Model<Project>,
+        save_intent: SaveIntent,
+        cx: &mut ViewContext<Self>,
+    ) -> Task<Result<()>> {
         self.report_editor_event("save", None, cx);
         let buffers = self.buffer().clone().read(cx).all_buffers();
         cx.spawn(|this, mut cx| async move {
             this.update(&mut cx, |this, cx| {
-                this.perform_format(project.clone(), FormatTrigger::Save, cx)
+                this.perform_format(
+                    project.clone(),
+                    if save_intent == SaveIntent::SaveWithoutFormatting {
+                        FormatTrigger::SaveWithoutFormatting
+                    } else {
+                        FormatTrigger::Save
+                    },
+                    cx,
+                )
```

Comment:
See comment below: why not do the check outside of the call to `.perform_format`? i.e.

```rust
if save_intent != SaveIntent::SaveWithoutFormatting {
                this.perform_format(
                    project.clone(),
                    FormatTrigger::Save,
                    cx,
                )
}